src/batch/src/task/task_execution.rs

```diff
@@ -433,14 +436,17 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
             serde_json::to_string_pretty(self.plan.get_root()?).unwrap()
         );
 
-        let exec = ExecutorBuilder::new(
-            self.plan.root.as_ref().unwrap(),
-            &self.task_id,
-            self.context.clone(),
-            self.epoch.clone(),
-            self.shutdown_rx.clone(),
+        let exec = captured_execution_context_scope!(
+            captured_execution_context.clone(),
```

Comment:
> As a result, we have to find all paths to provide the context before evaluating. ...(the frontend inline rewriting approach) is really easy to implement and not invasive at all.

We either need to find all paths to provide the context before evaluating, or find all paths to inline the context before evaluating. The primary goal of these `make_timestamptz` PRs are not to support this new function for users, but to be the pioneer and see if the new approach is better than existing one.

@KeXiangWang How did you figure out these places to provide context in this PR? If some of them slip through, how easy are they exposed as a runtime error? My impression was that there are less places and easier to surface compared to the expr rewrite approach (recent addition #13724). But this can be wrong and simply due to we had more time to receive bugs on the rewrite approach.

I also agree inlining is less invasive, to the extend that it is solely in frontend. But if approach A requires awareness in 10 submodules in 1 module, while approach B requires awareness in 2 submodules in 3 parent modules each, it may also be acceptable to consider A as more invasive because 10 > 2 * 3. Again these numbers are made up arbitrarily and personally I feel the difference between 10 and 6 is not large enough to justify it.

Reply:
I went through the codes in frontend and run some test examples to help verify whether all of them are found. 

The good thing is when there's an expression been evaluated without context, it will throw an error like "context TIME_ZONE not found" and the error is shown in psql console, making it easy to recognize. For streaming mode, it will generate one log with the same error message and when we read from the MV, the result of that row is missed.

Another benefit is that we only need to find all the paths (to provide the context before evaluating) once. Once it's finished correctly with `make_timestamptz`, unless we have new paths, we don't need to care it in following the implementation of other expressions.